test/integration/uds_integration_test.cc

```diff
@@ -7,25 +7,56 @@
 namespace Envoy {
 
 INSTANTIATE_TEST_CASE_P(IpVersions, UdsIntegrationTest,
-                        testing::ValuesIn(TestEnvironment::getIpVersionsForTest()));
+                        testing::Combine(testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                                         testing::Bool()));
 
 TEST_P(UdsIntegrationTest, RouterRequestAndResponseWithBodyNoBuffer) {
+#if !defined(__linux__)
```

Comment:
Instead of needing to copy/paste this into every test case, how about changing the testing::Bool() above to be a sequence that is {true, false} on linux and {false} on other platforms, and having 1 test that validates that you can create one of these on linux, and EXPECT_THROW on other platforms?

Reply:
Hmm I'm not sure how to do the last part. If I understand correctly you propose to run a single test with abstract == true on Non-linux instead of all of them. But the instantiate macro will instanciate all tests no matter what.

Reply:
You'll have to play around to see what works, but you'll end up somewhere with a clause like:
```
#if defined(__linux__)
testing::Values(false, true)
#else
testing::Values(false)
#endif
```

At least I think something like that will work.  See https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md

Reply:
It works, but on top of that I can't add a "single" test that tests the throwing on non-linux.
That snipped will not test the throwing on non-linux. If I add another test for "throwing" I can't enable it for abstract sockets, because the instantiate macro has testing::Values(false). Or I need another way to enable abstract_namespaces, which needs additional code in the class or a separate test class, which is not "simpler" at the end.

Reply:
For the non-parameterized case, just add a standalone test case: 

```TEST(test case name, test name) {
#if defined(__linux__)
do the thing
#else
expect the throw
#endif
}
```

I think that, plus the #if defined on the Values to create the test cases should be the only preprocessor goo needed in the tests.

Reply:
Yes but do the thing requires a separate UdsIntegrationTest-like class, which is a bit strange since we already have a class that does the same thing with parameters.

Reply:
I think this test will be a simple "can I construct one of these with a string starting with '@'".  It doesn't need a class, or to be very complicated at all.

Reply:
Actually, on second look, you already have that test in address_impl_test.cc.